Apps/WebClient/src/ClientApp/src/components/modal/vaccineCardModal.vue

```diff
@@ -175,37 +164,35 @@ export default class VaccineCardModalComponent extends Vue {
         this.$bvModal.hide("covidImmunizationCard");
     }
 
-    private downloadPdf() {
-        this.isDownloading = true;
-        SnowPlow.trackEvent({
-            action: "download_card",
-            text: "COVID Card PDF",
-        });
-
-        this.retrieveVaccineRecord({ hdid: this.user.hdid })
-            .then((payload: CovidVaccineRecord) => {
-                if (!payload.loaded) {
-                    this.logger.error(
-                        "Vaccine record could not be retrieved at this time"
-                    );
-                } else if (
-                    payload.document.data === null ||
-                    payload.document.data.length == 0
-                ) {
-                    this.logger.error("Vaccine record is empty");
-                } else {
-                    const mimeType = payload.document.mediaType;
-                    const downloadLink = `data:${mimeType};base64,${payload.document.data}`;
-                    fetch(downloadLink).then((res) => {
+    private download() {
+        const printingArea: HTMLElement | null =
+            document.querySelector(".vaccine-card");
+
+        if (printingArea !== null) {
+            this.isDownloading = true;
+
+            SnowPlow.trackEvent({
+                action: "download_card",
+                text: "COVID Card PDF",
+            });
+
+            html2canvas(printingArea, {
+                scale: 2,
+                ignoreElements: (element) =>
+                    element.classList.contains("d-print-none"),
+            })
+                .then((canvas) => {
+                    const dataUrl = canvas.toDataURL();
+                    fetch(dataUrl).then((res) => {
                         res.blob().then((blob) => {
-                            saveAs(blob, "BCVaccineRecord.pdf");
+                            saveAs(blob, "BCVaccineCard.png");
                         });
                     });
-                }
-            })
-            .finally(() => {
-                this.isDownloading = false;
-            });
+                })
```

Comment:
Was the indentation intentional for the new code?  New code added does not seem to have the same column alignment as with the previously deleted code.

Reply:
Formatting and indentation is handled by prettier in the front-end.  I use the prettier extension in VS Code, so the files are auto-formatted upon save.